relnotes/makelayout.migration.md

```diff
@@ -0,0 +1,43 @@
+* `ocean.util.log.Config`
+
+    The `configureLogger`, `configureLoggers` overloads and `configureOldLoggers`
+    have gained an additional `makeLayout` parameter.
+    This parameter is used to instantiate a `Layout` from a name, allowing
+    users to have custom layout usable from their config files.
+    For example, a config entry would be:
+    ```
+    [LOG.myapp.supermodule]
+    level = info
+    file  = log/supermodule.log
+    file_layout = aquatic
+    console_layout = submarine
+    ```
+    And calling `configureOldLoggers` should be done as such:
+    ```
+    void myConfigureLoggers (
+        ClassIterator!(Config, ConfigParser) config,
+        MetaConfig m_config,
+        Appender delegate ( istring file, Layout layout ) file_appender,
+        bool use_insert_appender = false)
+    {
+        scope makeLayout = (cstring name)
+        {
+            if (name == "aquatic")
+                return new AquaticLayout;
+            if (name == "submarine")
+                return new SubmarineLayout;
+            return ocean.util.log.Config.newLayout(name);
+        }
+        ocean.util.log.Config.configureOldLoggers(config, m_config,
+            file_appender, use_insert_appender, makeLayout);
+    }
+    ```
+    Note that the previous example is kept short for simplicity. `makeLayout` gets the raw name,
+    and as such should make sure to make the comparison as loose as wanted (by lowercasing,
+    removing `"layout"` prefix/suffix, etc...).
+
+    In addition, the members `console_layout` and `file_layout` in the `Config` class
```

Comment:
I wonder, would it be possible to leave the default values as an empty string and add an extra check that converts `""` -> `"simple"` or `"date"` in the point where the fields are referred to, in `configureLogger`? Would that remove the need for this migration instruction and make this a pure new feature?